fvm/src/blockstore/buffered.rs

```diff
@@ -34,6 +34,19 @@ where
     pub fn into_inner(self) -> BS {
         self.base
     }
+
+    /// Flushes all blocks from the write cache to the provided blockstore,
+    /// regardless of whether they're reachable from any state root.
+    /// Unlike the standard flush() operation which only writes blocks connected
+    /// to the final state tree, this writes every block created during execution.
+    pub fn flush_all(&self) -> Result<()> {
+        log::info!(
+            "Flushing all ({}) cache blocks to blockstore",
+            self.write.borrow().len()
+        );
+        self.base
+            .put_many_keyed(self.write.borrow().iter().map(|(k, v)| (*k, v.as_slice())))
```

Comment:
nit: `flush` actually _removes_ them from this blockstore to avoid writing twice (just in case). I'd do that here as well. I.e., `put_many_keyed(self.write.borrow_mut().drain()...`.

Reply:
ahh good point, I've made it do this and added some tests to assert the behaviour of this vs flush()